src/main/CommandHandler.cpp

```diff
@@ -363,28 +363,53 @@ enum class Requirement
     REQUIRED_REQ
 };
 
-template <typename T>
-optional<T>
+optional<uint32>
 parseNumParam(std::map<std::string, std::string> const& map,
-              std::string const& key, T& val)
+              std::string const& key)
 {
     auto i = map.find(key);
     if (i != map.end())
     {
+        uint32 val;
         std::stringstream str(i->second);
         str >> val;
 
         // Throw an error if not all bytes were loaded into `val`
-        if (str.fail() || str.rdbuf()->in_avail() > 0)
+        if (str.fail() || !str.eof())
         {
             std::string errorMsg =
                 fmt::format("Failed to parse '{}' argument", key);
             throw std::runtime_error(errorMsg);
         }
-        return make_optional<T>(val);
+        return make_optional<uint32>(val);
     }
 
-    return nullopt<T>();
+    return nullopt<uint32>();
+}
+
+optional<uint32>
+parseNumParam(std::map<std::string, std::string> const& map,
```

Comment:
After looking at the changes overall, I don't think this contract makes sense to me:
we already have a `parseNumParam` that returns an `optional<T>` (that can be used for anything).

From the use cases point of view, I think it would probably make more sense to rename the optional version to `maybeParseNumParam` and add a dedicated method to deal with the non-optional case:
in this case, the method's signature would not return an optional, and there is no default value.

Then it probably makes sense to update the `maybeParseNumParam` that returns an optional take care of the default value (and have the other one call it with a default value internally).